examples/common/text_data.py

```diff
@@ -61,13 +60,14 @@ def __init__(self,
                  validate_hash: Optional[str] = None,
                  shuffle_seed: int = 9176,
                  num_canonical_nodes: Optional[int] = None,
-                 batch_size: Optional[int] = None):
+                 batch_size: Optional[int] = None,
+                 **kwargs: Dict[str, Any]):
 
-        # Validation
-        if group_method not in ['truncate', 'concat']:
-            raise ValueError(
-                f"group_method='{group_method}' must be one of ['truncate', 'concat']."
-            )
+        if 'group_method' in kwargs:
+            raise DeprecationWarning(
+                'group_method is deprecated and has been removed.\nTo ' +
+                'concatenate, use the --concat_text or --concat_tokens ' +
+                'argument to concat_c4.py when creating your MDS dataset')
```

Comment:
python has deprecated warnings.
In python you can raise errors.
Can you raise a warning?

Is this valid python code? At the very least it seems like it [shouldn't](https://stackoverflow.com/a/29378596) be valid.

Reply:
It is a...choice. Normally we would warn a `DeprecationWarning` and then later remove it, but in this case we _have_ to completely get rid of the option, because the previous code does not work with the latest streaming version. So this is a way for us to indicate that we deprecated something and explain it to the user, but also completely remove the option without a period of deprecation first. I'd be fine making it a `ValueError` or something though.

Reply:
Maybe just say:
```
            raise NotImplementedError(
                'group_method is is no longer an option.\nTo ' +
                'concatenate, use the --concat_text or --concat_tokens ' +
                'argument to concat_c4.py when creating your MDS dataset'
```

Reply:
Yeah, that works for me.

Reply:
We should also maybe check that there are not other kwargs that the user is trying to pass to the fn ie

```
kwargs.pop('group_method', None)  # remove the one possible kwarg
if kwargs:
    raise TypeError(f'StreamingTextDataset() got an unexpected keyword argument: {kwargs}')
```